src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java

```diff
@@ -372,11 +370,10 @@ public String getGitHubAuthId() {
     
     public GhprbGitHubAuth getGitHubApiAuth() {
         if (gitHubAuthId == null) {
-            for (GhprbGitHubAuth auth: getDescriptor().getGithubAuth()){
-                gitHubAuthId = auth.getId();
-                getDescriptor().save();
-                return auth;
-            }
+            GhprbGitHubAuth auth = getDescriptor().getGithubAuth().get(0);
```

Comment:
That would be true, except in the current behavior if there aren't any it will fall through this and call `getDescriptor().getGitHubAuth(gitHubAuthId);` below with a null value.  And if you look in that method it returns `getDescriptor().getGithubAuth().get(0);` if a null is passed.  So basically that ends up getting called no matter what.  I'll take another look here and see if there's a better way to do it but if that's a concern there's most likely a bug here already.